pkg/lib/cockpit-components-inline-notification.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import cockpit from 'cockpit';
+
+import { Alert, AlertActionCloseButton } from "@patternfly/react-core/dist/esm/components/Alert/index.js";
+import type { AlertProps } from "@patternfly/react-core/dist/esm/components/Alert/index.js";
+import { Button } from "@patternfly/react-core/dist/esm/components/Button/index.js";
+import './cockpit-components-inline-notification.css';
+
+const _ = cockpit.gettext;
+
+export const InlineNotification = ({ text, detail, type = "danger", onDismiss, isInline = true, isLiveRegion = false }: {
+    text: string;
+    detail?: string;
+    type: AlertProps["variant"];
+    onDismiss?: (ev?: Event) => void;
+    isInline: boolean;
+    isLiveRegion: boolean;
+}) => {
+    const [isDetail, setIsDetail] = useState(false);
+
+    const detailButton = (detail &&
+        <Button variant="link" isInline className="alert-link more-button"
+            onClick={event => {
+                if (event.button !== 0)
+                    return;
+                event.preventDefault();
+                setIsDetail(prev => !prev);
+            }}
+        >
+            {isDetail ? _("show less") : _("show more")}
+        </Button>
+    );
+
+    const extraProps = {};
+    if (onDismiss)
+        // @ts-expect-error: PF bug: actionClose not defined in Alert
+        extraProps.actionClose = <AlertActionCloseButton onClose={onDismiss} />;
```

Comment:
This added line is not executed by any test.

Reply:
Can you not write this inline like:

```jsx
<Alert
  etc
  ...{ onDismiss && <AlertActionClodeButton onClose={onDismiss} /> }
  etc
  >
```

?

Does that help with the types at all?

Reply:
Yes, it does. Done.